cosmwasm/packages/wasmi-runtime/src/registration/cert.rs

```diff
@@ -347,6 +341,40 @@ pub fn verify_ra_cert(cert_der: &[u8]) -> SgxResult<Vec<u8>> {
     Ok(report_public_key)
 }
 
+#[cfg(all(feature = "SGX_MODE_HW", feature = "production"))]
+fn verify_quote_status(quote_status: SgxQuoteStatus) -> Result<(), sgx_status_t> {
+    match quote_status {
+        SgxQuoteStatus::OK => Ok(()),
+        SgxQuoteStatus::SwHardeningNeeded => {
+            warn!("Attesting enclave is vulnerable, and should be patched");
+            Ok(())
+        }
```

Comment:
wait, we DON'T want to crash on this in production??

Reply:
@Cashmaney you got blamed 😛 
Should this really give a warning  on production or should we crash instead?

Reply:
Intel's documentation on this status, for reference:
```
"SWHardeningNeeded" - TCB level of the SGX platform is up-to-date but due to certain issues affecting the platform, additional SW Hardening in the attesting SGX enclaves may be needed.
```
https://api.portal.trustedservices.intel.com/documentation

or -
```
SW_HARDENING_NEEDED – the EPID signature of the ISV enclave QUOTE has been verified correctly but due to certain issues affecting the platform, additional SW Hardening in the attesting SGX enclaves may be needed. The relying party should evaluate the potential risk of an attack leveraging the relevant issues on the attesting enclave, and whether the attesting enclave employs adequate software hardening to mitigate the risk.
```
https://api.trustedservices.intel.com/documents/sgx-attestation-api-spec.pdf

Reply:
Based on this:
https://cacheoutattack.com/files/SGAxe.pdf - section *Attestation Protocol Versions*
> As Intel has no way of
> remotely verifying if such hardening was perform or not, on
> machines vulnerable to LVI (nearly all currently available Intel
> machines), version 4 of the attestation protocol outputs SW -
> HARDENING NEEDED which indicates that our malicious
> quoting enclave’s “has been verified correctly but due to certain issues affecting the platform, additional SW Hardening in
> the attesting SGX enclaves may be needed” [48, Page 22]. Intel
> further instructs “the relying party should evaluate...whether
> the attesting enclave employs adequate software hardening
> to mitigate the risk”. This indicates that Intel asks enclave
> providers to trust their own enclaves only if they have performed LVI mitigations [42], without being able to remotely
> verify so. 

I'd say we don't want that in production.

Reply:
Right now there's no microcode fix for the LVI issue, and so it will always return SW_HARDENING_NEEDED

Reply:
Ahh, so SW in this context always refers to our code then?
Sounds legit. Maybe we should print the Advisory IDs as well in the warning.